src/Form/DataTransformer/ArrayToModelTransformer.php

```diff
@@ -59,6 +68,9 @@ public function reverseTransform($array)
         return $this->modelManager->modelReverseTransform($this->className, $array);
     }
 
+    /**
+     * @phpstan-param T $value
```

Comment:
For the same reason than https://github.com/sonata-project/SonataAdminBundle/pull/6377/files#r483949650,
I feel like the phpdoc should be `T|null`.

And we should also add `@param object|null $value`